src/test/java/frc/team997/lib/Trajectory/AsymmetricTrapezoidProfileTests.java

```diff
@@ -0,0 +1,151 @@
+                new State(1.605, 1.1)
+        );
+        assertEquals(
+                positiveTriangleProfile.calculate(1.5),
+                new State(1.9461524227066316, 0.4641016151377544)
+        );
+        assertEquals(
+                positiveTriangleProfile.calculate(2.0),
+                new State(2.0, 0.0)
+        );
+    }
+    @Test
+    public void asymmetricPositiveTriangleTimeLeftUntil() {
+        assertEquals(
+                positiveTriangleProfile.timeLeftUntil(1),
+                0.0, epsilon
+        );
+        assertEquals(
+                positiveTriangleProfile.timeLeftUntil(1.92),
+                1.4492, epsilon
+        );
+    }
+    @Test
+    public void asymmetricPositiveTriangleIsFinished() {
+        assertFalse(positiveTriangleProfile.isFinished(1.2));
+        assertTrue(positiveTriangleProfile.isFinished(2.0));
+    }
+
+    AsymmetricTrapezoidProfile negativeTrapezoidProfile = new AsymmetricTrapezoidProfile(
+            new Constraints(2, 1, -2),
+            new State(-1, 0),
+            new State(3, 0)
+    );
+    @Test
+    public void asymmetricNegativeTrapezoidPhases() {
+        var expected = new ArrayList<>(List.of(
+                new Phase(2.0, -2.0, -1.0, 0.0),
+                new Phase(0.5, -1.0, 0.0, -2.0),
+                new Phase(1.0, -1.0, 2.0, -2.0)
+        ));
+        assertEquals(negativeTrapezoidProfile.phases, expected);
+    }
+    @Test
+    public void asymmetricNegativeTrapezoidTotalTime() {
+        assertEquals(
+                negativeTrapezoidProfile.totalTime(),
+                3.5, epsilon
+        );
+    }
+    @Test
+    public void asymmetricNegativeTrapezoidCalculate() {
+        assertEquals(
+                negativeTrapezoidProfile.calculate(0),
+                new State(3.0, 0.0)
+        );
+        assertEquals(
+                negativeTrapezoidProfile.calculate(0.5),
+                new State(2.875, -0.5)
+        );
+        assertEquals(
+                negativeTrapezoidProfile.calculate(1.1),
+                new State(2.395, -1.1)
+        );
+        assertEquals(
+                negativeTrapezoidProfile.calculate(3.0),
+                new State(-0.75, -1.0)
+        );
+        assertEquals(
+                negativeTrapezoidProfile.calculate(3.5),
+                new State(-1.0, 0.0)
+        );
+    }
+    @Test
+    public void asymmetricNegativeTrapezoidTimeLeftUntil() {
+        assertEquals(
+                negativeTrapezoidProfile.timeLeftUntil(1.92),
+                1.469693845669907, epsilon
+        );
+        assertEquals(
+                negativeTrapezoidProfile.timeLeftUntil(1),
+                2.0, epsilon
+        );
+    }
+    @Test
+    public void asymmetricNegativeTrapezoidIsFinished() {
+        assertFalse(negativeTrapezoidProfile.isFinished(1.2));
+        assertTrue(negativeTrapezoidProfile.isFinished(3.5));
+    }
+    // TODO cases with too steep accel/decel
```

Comment:
Is this case just untested, or unimplemented?

Reply:
it’s tested and all cases are working, just haven’t gotten around to writing the unit tests for some cases